### PR TITLE
feat(eslint-plugin): extend eslint semi rule

### LINKED ISSUE #366: [semi] Semicolons aren't checked for type aliases
**Repro**

```TS
type Foo = {}
type Bar = {};
```

**Expected Result**
One of the lines should be wrong, but they're both valid, regardless of ESLint config.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.1` |
| `@typescript-eslint/parser`        | `1.4.1` |

### PR DESCRIPTION
**Pull request by aprilrd**
Fixed #366, #123 and #409.

~I plan on fixing #123 and #409 in this pr as well.~

### CODE REVIEW

File: packages/eslint-plugin/ROADMAP.md [hunk 1]
```diff
@@ -177,7 +177,7 @@
 | [`prefer-while`]                    | 🛑  | N/A                                                                                 |
 | [`quotemark`]                       | 🌟  | [`quotes`][quotes]                                                                  |
 | [`return-undefined`]                | 🛑  | N/A                                                                                 |
-| [`semicolon`]                       | 🌟  | [`semi`][semi]                                                                      |
+| [`semicolon`]                       | 🌓  | [`@typescript-eslint/semi`]                                                         |
```
Comment (bradzacher) [inline packages/eslint-plugin/ROADMAP.md]: you have to add the reference to the list at the bottom of the file, or else it won't create a link.
image

**Comment by JamesHenry** [comment]
Wait, @bradzacher what happened here?

I went down PRs in reverse chronological order (probably shouldn't have done that but beside the point), and I have just merged a PR from you for this exact rule, despite you have having approved an already open PR?

**Comment by JamesHenry** [comment]
I'm sorry for the confusion here @aprilrd I had no intention of missing our your contribution, thanks so much for taking the time to do it

**Comment by aprilrd** [comment]
It is a bummer. If you would be so kind, I would like to understand the issues with this PR so that I can do better next time.

**Comment by bradzacher** [comment]
Ahh crap, I'm so sorry about this.
You did nothing wrong @aprilrd, this was 100% my fault; I wasn't paying enough attention when going through the issue log.

I was going through adding fixes for issues tagged as bug, noticed 3 issues for semi, and prioritised it; completely forgetting that there was this open PR that I'd already reviewed a few weeks ago.

If you're willing - there's a few places that your PR can still contribute to this - specifically:
- your much, much more comprehensive tests,
- your update to `packages/eslint-plugin/ROADMAP.md`

Again, I'm sorry that I slipped up here.

**Comment by aprilrd** [comment]
I understand. I will make a separate pr with the readme and the test.

**Pull request closed without merging**